### PR TITLE
[BALDE-3] File Manager

### PR DESCRIPTION
**Pull request by Blinkuu**
Implementation of File Manager class. BIG file reading for Windows not done yet. Passes all tests.

### CODE REVIEW

File: src/utils/file_manager.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
+//
+// Created by blinku on 15.12.18.
+//
+
+#pragma once
+
+#include "pch.h"
```
Comment (BIGbadEL) [inline src/utils/file_manager.h]: precompile header should be included in source file

File: src/utils/file_manager.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        enum Size : char {
+            SMALL, BIG
+        };
+    public:
+
+        /**
+         * @fn FileManager
+         * @brief default constructor
+         */
+        FileManager() = default;
```
Comment (BIGbadEL) [inline src/utils/file_manager.h]: We shouldn't be ablle to create object of this type

File: src/utils/file_manager.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+        ~FileManager() = default;
+
+        /**
+         * @fn ReadFile
+         * @brief use to read files
+         * @return string
+         */
+
+        static std::string ReadFile(const char* filePath, Size size);
```
Comment (BIGbadEL) [inline src/utils/file_manager.h]: should be [[nodiscard]]

File: src/utils/file_manager.h [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    private:
+        static std::string ReadSmallFile(const char* filePath); /** used to read small files, same implementation on windows and linux*/
```
Comment (BIGbadEL) [inline src/utils/file_manager.h]: should be [[nodiscard]]

File: src/utils/file_manager.h [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        static std::string ReadBigFile(const char* filePath); /** used to read big files, different implementation for windows and linux*/
```
Comment (BIGbadEL) [inline src/utils/file_manager.h]: should be [[nodiscard]]

File: tests/test_file_manager.cpp [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+// Created by blinku on 16.12.18.
+//
+
+#include "pch.h"
+#include "gtest/gtest.h"
+#include "utils/file_manager.h"
+
+TEST(FileManager, GoodSmallFileOpening) { //NOLINT
+
+    std::string file_result = Bald::Utils::FileManager::ReadFile("/home/blinku/CLionProjects/BALD/bald-engine/tests/test_file_manager.txt", Bald::Utils::FileManager::SMALL);
```
Comment (BIGbadEL) [inline tests/test_file_manager.cpp]: this path should't be relative to your system

File: tests/test_file_manager.cpp [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+    EXPECT_EQ("plik testowy\n"
+              "do odczytu", file_result);
+}
+
+TEST(FileManager, GoodBigFileOpening) { //NOLINT
+
+    std::string file_result = Bald::Utils::FileManager::ReadFile("/home/blinku/CLionProjects/BALD/bald-engine/tests/test_file_manager.txt", Bald::Utils::FileManager::BIG);
```
Comment (BIGbadEL) [inline tests/test_file_manager.cpp]: this path should't be relative to your system

File: tests/test_file_manager.cpp [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+    EXPECT_EQ("plik testowy\n"
+              "do odczytu", file_result);
+}
+
+
+TEST(FileManager, WrongSmallFileOpening) { //NOLINT
+
+    std::string file_result = Bald::Utils::FileManager::ReadFile("no_such_file.cpp", Bald::Utils::FileManager::SMALL);
```
Comment (BIGbadEL) [inline tests/test_file_manager.cpp:28]: this path should't be relative to your system

File: CMakeLists.txt [hunk 9]
```diff
@@ -16,13 +17,28 @@ add_compile_options(
...
\ No newline at end of file
+cotire(bald_engine)
+
+enable_testing()
+#file(GLOB TEST_SRC_FILES ${PROJECT_SOURCE_DIR}/tests/*.cpp)
+
+message("-- TEST EXECUTALES: " ${TEST_SRC_FILES})
+#add_executable(runUnitTests ${TEST_SRC_FILES})
+
+add_executable(runUnitTests
```
Comment (BIGbadEL) [inline CMakeLists.txt]: name of the test should be somthing like file_menager_test

File: src/utils/file_manager.cpp [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+#include <sys/mman.h>
+
+namespace Bald::Utils {
+
+    std::string FileManager::ReadFile(const char *filePath, Size size) {
+        if (size == SMALL) {
+            return ReadSmallFile(filePath);
+        } else if (size == BIG) {
+            return ReadBigFile(filePath);
+        } else {
```
Comment (BIGbadEL) [inline src/utils/file_manager.cpp]: that else is unnecessary

File: src/utils/file_manager.cpp [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+    std::string FileManager::ReadSmallFile(const char *filePath) {
+        FILE *file = fopen(filePath, "r");
+
+        if (!file) {
+            CORE_LOG_WARN("Couldn't open the file at path: ", filePath);
+            return std::string("Error!");
+        }
+
+        fseek(file, 0, SEEK_END);
+        unsigned long stringSize = ftell(file);
```
Comment (BIGbadEL) [inline src/utils/file_manager.cpp]: stringSize should be long and could be const

File: src/utils/file_manager.cpp [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+
+        delete[] buffer;
+
+        fclose(file);
+
+        return result;
+    }
+
+    std::string FileManager::ReadBigFile(const char *filePath) {
+#define _LINUX
```
Comment (BIGbadEL) [inline src/utils/file_manager.cpp]: this define should be in cmake or test or sample not in source file!!!

File: src/utils/file_manager.cpp [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+namespace Bald::Utils {
+
+    std::string FileManager::ReadFile(const char *filePath, Size size) {
+        if (size == SMALL) {
+            return ReadSmallFile(filePath);
+        } else if (size == BIG) {
+            return ReadBigFile(filePath);
+        }
+        CORE_LOG_WARN("Specify size of your file! (SMALL/BIG)");
```
Comment (BIGbadEL) [inline src/utils/file_manager.cpp:19]: code after this if is never gonna be reached

File: src/utils/file_manager.h (hunk 1 above)
Comment (Blinkuu) [reply src/utils/file_manager.h]: moved to src

File: src/utils/file_manager.h (hunk 2 above)
Comment (Blinkuu) [reply src/utils/file_manager.h]: done

File: src/utils/file_manager.h (hunk 3 above)
Comment (Blinkuu) [reply src/utils/file_manager.h]: done

File: src/utils/file_manager.h (hunk 4 above)
Comment (Blinkuu) [reply src/utils/file_manager.h]: done

File: src/utils/file_manager.h (hunk 5 above)
Comment (Blinkuu) [reply src/utils/file_manager.h]: done

File: tests/test_file_manager.cpp (hunk 6 above)
Comment (Blinkuu) [reply tests/test_file_manager.cpp]: done

File: tests/test_file_manager.cpp (hunk 7 above)
Comment (Blinkuu) [reply tests/test_file_manager.cpp]: done

File: tests/test_file_manager.cpp (hunk 8 above)
Comment (Blinkuu) [reply tests/test_file_manager.cpp:28]: done

File: CMakeLists.txt (hunk 9 above)
Comment (Blinkuu) [reply CMakeLists.txt]: done

File: src/utils/file_manager.cpp (hunk 10 above)
Comment (Blinkuu) [reply src/utils/file_manager.cpp]: done

**Pull request closed without merging**